pkg/logger/logger.go

```
@@ -0,0 +1,28 @@
+package logger
+
+// ContainerLogger defines a logging backend
+type ContainerLogger interface {
+	// Send stdout line to the backend
+	Stdout(line string) error
+
+	// Send stderr line to the backend
+	Stderr(line string) error
+
+	// Close stdout handler
+	CloseStdout()
```

Is there any reason you want to have a separate method to close stdout and stderr ? We will always want to close both stream no ?

---

I would also return an error here. So we can at least log the error if any.

---

Since it's 2 different handler for reading them, I keep both for closing. I guess they will always close at the same time, but nothing guarantee it. And for redis, internally we support two two channels for stdout/stderr, we could also support two files for stdout/stderr, which are not the same, etc. I guess it's better to keep separation ?

---

Handled by io.Writer now